helm-charts/basehub/values.yaml

```diff
@@ -400,7 +404,77 @@ jupyterhub:
                   - value: "/rstudio"
                     title: RStudio
                     description: An IDE For R, created by the RStudio company
-
+    initContainers:
+      - name: templates-clone
+        image: alpine/git
+        args:
+          - clone
+          - --
+          - $(HUB_TEMPLATES_REPO)
+          - /srv/repo
+        env:
+          - name: HUB_TEMPLATES_REPO
+            valueFrom:
+              configMapKeyRef:
+                name: hub-custom-templates-config
+                key: HUB_TEMPLATES_REPO
+        securityContext:
+          runAsUser: 1000
+          runAsGroup: 1000
+          allowPrivilegeEscalation: False
+          readOnlyRootFilesystem: True
+        volumeMounts:
+          - name: custom-templates
+            mountPath: /srv/repo
+      - name: templates-ownership-fix
+        image: alpine/git
+        command:
+          - /bin/sh
+        args:
+          - -c
+          - ls -lhd /srv/repo && chown 1000:1000 /srv/repo && ls -lhd /srv/repo
+        securityContext:
+          runAsUser: 0
+        volumeMounts:
+          - name: custom-templates
+            mountPath: /srv/repo
+    extraContainers:
+      - name: templates-sync
+        image: alpine/git
+        workingDir: /srv/repo
+        command:
+          - /bin/sh
+        args:
+          - -c
+          - |
+            ls -lhd /srv/repo;
+            while true; do git fetch origin;
+            git reset --hard origin/$(HUB_TEMPLATES_REPO_BRANCH);
```

Comment:
To simplify containers setup, we will be trading off automatically checking if a branch with `<cluster-name>-<hub-name>` exists and pull from there for a given hub. Instead, we will need to specify the branch in each hub's config. See the 2i2c-ohw config in this PR.

But I believe this is super ok, as it's always better to be explicit and the template containers configs gets to stay in one central place (the basehub)